EssentialsProtect/src/com/earth2me/essentials/protect/EssentialsProtectBlockListener.java

```diff
@@ -58,6 +62,37 @@ public void onBlockIgnite(BlockIgniteEvent event) {
         }
     }
 
+    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
+    public void onEntityDamageByEntity(EntityDamageByEntityEvent event) {
+        if (event.getEntity() instanceof Item) {
+            if (event.getCause() == EntityDamageEvent.DamageCause.FIRE || event.getCause() == EntityDamageEvent.DamageCause.FIRE_TICK || event.getCause() == EntityDamageEvent.DamageCause.LAVA) {
+                event.setCancelled(prot.getSettingBool(ProtectConfig.disable_lava_item_dmg));
+            }
+
+            if (event.getCause() == EntityDamageEvent.DamageCause.ENTITY_EXPLOSION) {
+                if (event.getDamager() instanceof TNTPrimed) {
+                    event.setCancelled(prot.getSettingBool(ProtectConfig.prevent_tnt_itemdmg));

```

Comment:
So this `event.setCancelled(prot.getSettingBool(ProtectConfig)` line gets repeated a bunch. This should be cleaned up so that it reduces this code duplication.

Reply:
I'm not exactly sure what you want me to do. The same thing happens in the block ignite event right above my event. Any brilliant ideas?

Reply:
I'm not sure what @Ichbinjoe is proposing either.

My main concern with this is that if another plugin happens to have blocked something from happening, this might now uncancel events, but I'll check through this PR again when I get a chance.

Reply:
Ah I didn't look above to see the current style. Seeing as this is the current style of this file its okay, but I think my suggestion was going to be more along the lines of the following:

```
ProtectConfig configNode = null;
if (event.getDamager() instanceof Something) {{
    configNode = ProtectConfig.something;
} else if ...

if (configNode != null) {
    event.setCancelled(prot.getSettingBool(configNode));
}
```
However at this point it is more of a style & maintainability nitpick than anything.

On a separate note, since these are disjunct cases they should be else if, not just if. (Same with the above, but at this point that kind of got slipped through)